### PR TITLE
server: Verify status code on update check

### LINKED ISSUE #6687: server: Error decoding updates info: invalid character 'h' looking for beginning of value
```
W160515 09:37:13.122370 server/updates.go:203  Error decoding updates info: invalid character 'h' looking for beginning of value
```

Occasionally shows up when running

```
make build && ( killall -9 cockroach; rm -rf cockroach-data && ./cockroach start --background --alsologtostderr=INFO )
```

Probably a timing issue where we query some endpoint before it's ready.

### PR DESCRIPTION
**Pull request by nvb**
Closes #6687.

Previously we would try to JSON decode a response body without checking
its status code. This led to log errors like:

```
Error decoding updates info: invalid character 'h' looking for beginning
of value
```

Now we get something like:

```
Error checking for updates: 500 Internal Server Error: http.go:96:
failed to fetch updates for UUID c4e50fd6-02cf-4605-8b61-49410c0ace11:
dial tcp 127.0.0.1:26257: socket: too many open files
```

Related to https://github.com/cockroachlabs/registration/issues/25.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7543)

### CODE REVIEW

**Comment by dt** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543#-:-KLUt16IC6FE9VLmCINv:1417088721)*

File: server/updates.go [hunk 1]
```diff
@@ -182,6 +183,16 @@ func (s *Server) checkForUpdates() {
 	}
 	defer res.Body.Close()
 
+	if res.StatusCode != http.StatusOK {
+		b, err := ioutil.ReadAll(res.Body)
+		if err != nil {
```
Comment (tamird) [inline server/updates.go]: nit:why not log all the things?
```
if resp.StatusCode != http.StatusOK {
	b, err := ioutil.ReadAll(resp.Body)
	log.Warningf("status: %s, body: %s, error: %v", resp.Status, b, err)
	return
}
```

**Comment by tamird** [comment]
LGTM

**Comment by tamird** [comment]
This should close #6687.

File: server/updates.go (hunk 1 above)
Comment (nvanbenschoten) [inline server/updates.go]: I figured it was useful to mention "Error checking for updates" to give some context to the log, and "Error checking for updates, ..., error: nil" would look kind of silly.

Comment (tamird) [inline server/updates.go]: There are other words you can use. "Failed to check for updates", perhaps?

**Comment by tamird** [comment]
By the way, `reportUsage` similarly ignores status codes, mind fixing that while you're here?

File: server/updates.go (hunk 1 above)
Comment (nvanbenschoten) [inline server/updates.go]: Done.

**Comment by nvanbenschoten** [comment]
Done.

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[server/updates.go, line 286 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543#-KLe-kzYBGpOUjY5YglN:-KLe-kzYBGpOUjY5YglO:-1697233959) ([raw file](https://github.com/cockroachdb/cockroach/blob/633a0a340f446deb084f298470144243bde99b6a/server/updates.go#L286)):*
> ```Go
> 	if res.StatusCode != http.StatusOK {
> 		b, err := ioutil.ReadAll(res.Body)
> 		log.Warningf("Failed checking reporting node usage metrics: status: %s, body: %s, "+
> ```

here and above: "Failed to report"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543#-:-KLe03BuD0am5yjeHPCv:1417088721)*

**Comment by nvanbenschoten** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[server/updates.go, line 286 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543#-KLe-kzYBGpOUjY5YglN:-KLe3ki2iv5j-8W8CQKE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/633a0a340f446deb084f298470144243bde99b6a/server/updates.go#L286)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> here and above: "Failed to report"
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7543)*

**Pull request merged**